src/runtime/storageNG/tests/store-test.ts

```diff
@@ -0,0 +1,231 @@
+/**
+ * @license
+ * Copyright (c) 2019 Google Inc. All rights reserved.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * Code distributed by Google as part of this project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+import {assert} from '../../../platform/chai-web.js';
+import {Store, StorageMode, DirectStore, ProxyMessageType} from '../store.js';
+import {Exists, DriverFactory, StorageDriverProvider, Driver, ReceiveMethod} from '../drivers/driver-factory.js';
+import {CRDTCount, CountOpTypes, CountData, CountOperation} from '../../crdt/crdt-count.js';
+
+class MockDriver<Data> extends Driver<Data> {
+  receiver: ReceiveMethod<Data>;
+  async read(key: string) { throw new Error("unimplemented"); }
+  async write(key: string, value: {}) { throw new Error("unimplemented"); }
+  registerReceiver(receiver: ReceiveMethod<Data>) {
+    this.receiver = receiver;
+  }
+  async send(model: Data): Promise<boolean> {
+    throw new Error("send implementation required for testing");
+  }
+}
+
+class MockStorageDriverProvider implements StorageDriverProvider {
+  willSupport(storageKey: string) {
+    return true;
+  }
+  driver<Data>(storageKey: string, exists: Exists): Driver<Data> {
+    return new MockDriver<Data>(storageKey, exists);
+  }
+}
+
+describe('Store', async () => {
+
+  it(`will throw an exception if an appropriate driver can't be found`, async () => {
```

Comment:
This should exercise the willSupport logic when a driver has been registered but doesn't accept the given key.

Reply:
That's outside the scope of unit tests of Store though. Perhaps I can test that separately when driver infrastructure is more fleshed out, or as a simple integration test, or something.